src/AbstractFixer.php

```diff
@@ -71,9 +72,50 @@ final public function fix(\SplFileInfo $file, Tokens $tokens): void
             throw new RequiredFixerConfigurationException($this->getName(), 'Configuration is required.');
         }
 
+        $changed = $tokens->isChanged();
+
+        $isIgnored = false;
+        $tokenMapping = [];
+        foreach ($tokens as $index => $token) {
+            if (!$isIgnored && $this->tokenDisablesFixer($token)) {
+                $isIgnored = true;
+
+                continue;
+            }
+
+            if ($isIgnored && $token->isGivenKind(T_COMMENT) && '// phpcsfixer:enable' === $token->getContent()) {
+                $isIgnored = false;
+
+                continue;
+            }
+
+            if ($isIgnored) {
+                $dummyToken = new Token([999999, $token->getContent()]);
```

Comment:
this is extremely dangerous. here, the invalid token is created. It is affecting traversing logic of each rules, as they do things like `->getNextTokenOfKind(KIND_ID)`, and may crash if not found.

I'm unable to, and see it as huge effort, to ensure each rule can handle invalid token in any part of codebase and still work properly with it.

Reply:
example 1:

```php
<?php

array(
    1   ,
    2   ,
// phpcsfixer:disable
);
// phpcsfixer:enable
```

```
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ ./php-cs-fixer check --diff x.php --rules no_whitespace_before_comma_in_array -vvv
PHP CS Fixer 3.41.2-DEV Long Night by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.3.0
Loaded config default from "/Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/.php-cs-fixer.dist.php".
Paths from configuration file have been overridden by paths provided as command arguments.
E                                                                                                                                1 / 1 (100%)
Legend: .-no changes, F-fixed, S-skipped (cached or empty file), I-invalid file syntax (file ignored), E-error

Found 0 of 1 files that can be fixed in 0.003 seconds, 14.000 MB memory used

Files that were not fixed due to errors reported during fixing:
   1) /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/x.php

                                    
        [UnexpectedValueException]  
        Missing block "end".        
                                    

      PhpCsFixer\Tokenizer\Tokens->findOppositeBlockEdge()
        in /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/src/Tokenizer/Tokens.php at line 445
      PhpCsFixer\Tokenizer\Tokens->findBlockEnd()
        in /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/src/Fixer/ArrayNotation/NoWhitespaceBeforeCommaInArrayFixer.php at line 89
      PhpCsFixer\Fixer\ArrayNotation\NoWhitespaceBeforeCommaInArrayFixer->fixSpacing()
        in /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/src/Fixer/ArrayNotation/NoWhitespaceBeforeCommaInArrayFixer.php at line 64
      PhpCsFixer\Fixer\ArrayNotation\NoWhitespaceBeforeCommaInArrayFixer->applyFix()
        in /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/src/AbstractFixer.php at line 104
      PhpCsFixer\AbstractFixer->fix()
        in /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/src/Runner/Runner.php at line 173
      PhpCsFixer\Runner\Runner->fixFile()
        in /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/src/Runner/Runner.php at line 114
      PhpCsFixer\Runner\Runner->fix()
        in /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/src/Console/Command/FixCommand.php at line 302
      PhpCsFixer\Console\Command\FixCommand->execute()
        in /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/vendor/symfony/console/Command/Command.php at line 326
      [ ... ]

```

Reply:
example 2:
```php
<?php

array(
    1,
    2,
    3,
    // phpcsfixer:disable
    array(
        4,
        5
    // phpcsfixer:enable
    ),
    6
);
```
```
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ ./php-cs-fixer check --diff x.php --rules array_syntax -vvv
PHP CS Fixer 3.41.2-DEV Long Night by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.3.0
Loaded config default from "/Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/.php-cs-fixer.dist.php".
Paths from configuration file have been overridden by paths provided as command arguments.
E                                                                                                                                1 / 1 (100%)
Legend: .-no changes, F-fixed, S-skipped (cached or empty file), I-invalid file syntax (file ignored), E-error

Found 0 of 1 files that can be fixed in 0.005 seconds, 14.000 MB memory used

Files that were not fixed due to errors reported during linting after fixing:
   1) /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/x.php

                                                                            
        [PhpCsFixer\Linter\LintingException]                                
        Parse error: Unclosed '(' on line 8 does not match ']' on line 12.  
                                                                            

      Applied fixers: array_syntax
      ---------- begin diff ----------
--- /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/x.php
+++ /Users/d.ruminski/github/FriendsOfPHP/PHP-CS-Fixer/x.php
@@ -1,6 +1,6 @@
 <?php
 
-array(
+[
     1,
     2,
     3,
@@ -9,6 +9,6 @@
         4,
         5
     // phpcsfixer:enable
-    ),
+    ],
     6
 );
\ No newline at end of file

      ----------- end diff -----------
```